src/studio-home/tabs-section/courses-tab/courses-filters/courses-imported-filter-modal/context.tsx

```diff
@@ -0,0 +1,107 @@
+import {
+  createContext, ReactNode, useContext, useEffect, useMemo, useState,
+} from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useMigrationInfo } from '@src/library-authoring/data/apiHooks';
+import { getLoadingStatuses, getStudioHomeCoursesParams, getStudioHomeData } from '@src/studio-home/data/selectors';
+import { useLibraryContext } from '@src/library-authoring/common/context/LibraryContext';
+import { fetchStudioHomeData } from '@src/studio-home/data/thunks';
+import { updateStudioHomeCoursesCustomParams } from '@src/studio-home/data/slice';
+import { RequestStatus } from '@src/data/constants';
+
+export interface CourseImportFilterContextType {
+  processedMigrationInfo: Record<string, string[]>;
+  hidePreviouslyImportedCourses: boolean;
+  setHidePreviouslyImportedCourses: React.Dispatch<React.SetStateAction<boolean>>;
+  filteredCourses: object[];
+}
+
+export const CourseImportFilterContext = createContext<CourseImportFilterContextType | undefined>(undefined);
+
+interface Props {
+  children: ReactNode;
+  handleSelect?: (courseId: string) => void;
+  selectedCourseId?: string;
+}
+
+const PAGE_SIZE = 500;
+
+export const CourseImportFilterProvider = ({ handleSelect, selectedCourseId, children }: Props) => {
+  const { libraryId } = useLibraryContext(true) || {};
+  const dispatch = useDispatch();
+  const [hidePreviouslyImportedCourses, setHidePreviouslyImportedCourses] = useState(true);
+  const { courses, numPages } = useSelector(getStudioHomeData);
+  const studioHomeCoursesParams = useSelector(getStudioHomeCoursesParams);
+  const { currentPage } = studioHomeCoursesParams;
+  const { courseLoadingStatus } = useSelector(getLoadingStatuses);
+  const isCourseListLoading = courseLoadingStatus === RequestStatus.IN_PROGRESS;
+  const allVisibleCourseIds = courses?.map(item => item.courseKey) || [];
+  const {
+    data: migrationInfoData,
+    isPending,
+  } = useMigrationInfo(allVisibleCourseIds, allVisibleCourseIds.length > 0);
+
+  const processedMigrationInfo: Record<string, string[]> = useMemo(() => {
+    const result = {};
+    if (migrationInfoData) {
+      for (const libraries of Object.values(migrationInfoData)) {
+        // The map key in `migrationInfoData` is in camelCase.
+        // In the processed map, we use the key in its original form.
+        result[libraries[0]?.sourceKey] = libraries.map(item => item.targetKey);
```

Comment:
nit: to prevent a `result["undefined"] = []` item
```suggestion
        if (libraries.length !== 0) {
          result[libraries[0].sourceKey] = libraries.map(item => item.targetKey);
        }
```

Reply:
Done.